### PR TITLE
[pvr] refactor guide osd to be channel guide dialog and use it in guide info

### PR DESCRIPTION
**Pull request by xhaggi**
This PR refactor the PVR dialog guide OSD to supports listing of all EPG items for a supplied channel instead of just the current playing channel in fullscreen mode. Therefore the dialog is renamed to `GUIDialogPVRChannelGuide`.

I also added a new button _Channel guide_ to the guide info dialog from where you can open the full EPG list for this channel.

screenshot003

@ksooo mind taking a look. I hope you remember that we talked about it some times ago 😄 
@phil65 or @ronie It would be nice if you could adjust the skins.

### CODE REVIEW

File: addons/skin.estuary/xml/DialogPVRChannelGuide.xml
Comment (ksooo) [inline addons/skin.estuary/xml/DialogPVRChannelGuide.xml]: I suggest to remove the code instead of only commenting it.

File: xbmc/pvr/PVRGUIActions.h
Comment (ksooo) [inline xbmc/pvr/PVRGUIActions.h]: "must a " -> "must be a"

EDIT: despite from this, item can be either a channel, an epg tag or a timer. CPVRItem::GetChannel() takes care of the conversion. You are actually feeding this method with an epg tag from CGUIDialogPVRGuideInfo.

File: xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.h
Comment (ksooo) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.h]: While you are at refactoring this, would it make sense to use a std::unique_ptr instead of a raw pointer here?

File: xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp
Comment (ksooo) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp]: I know it was like this before, but it is no good idea to call into g_PVRManager with locked g_graphicsContext. May I suggest to move the two lines above to the very beginning of this method, before locking the global graphics mutex?

File: xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp
Comment (ksooo) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp]: m_channel.reset() is more efficient.

File: xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp
Comment (ksooo) [inline xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp]: Could you please add the text for the two resource identifiers as a comment, like we usually do?

**Review by ksooo — CHANGES_REQUESTED** [review]
Nice one, like it. Despite from my minor nitpicking it is fine already.

File: addons/skin.estuary/xml/DialogPVRChannelGuide.xml
Comment (xhaggi) [inline addons/skin.estuary/xml/DialogPVRChannelGuide.xml]: it's part of the skip commit. I don't want to merge it as it is, because it needs a bit more love from @ronie or @phil65

File: xbmc/pvr/PVRGUIActions.h
Comment (xhaggi) [inline xbmc/pvr/PVRGUIActions.h]: will adjust the doc

File: xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.h
Comment (xhaggi) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.h]: sure could be done in a separate commit

File: xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp
Comment (xhaggi) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp]: ok, will move it up.

File: xbmc/input/ButtonTranslator.cpp
Comment (da-anda) [inline xbmc/input/ButtonTranslator.cpp]: I think it would be nice if we'd keep this ID in addition to the new one for backwards compat reasons to not kill all custom user keymaps. We could add a comment to remove it in like 2 versions from now.

File: xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp
Comment (xhaggi) [inline xbmc/pvr/dialogs/GUIDialogPVRChannelGuide.cpp]: it does the same IIRC, but sure i can use `reset()` instead

File: xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp
Comment (xhaggi) [inline xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp]: c/p from another method. will change it.

File: xbmc/input/ButtonTranslator.cpp
Comment (xhaggi) [inline xbmc/input/ButtonTranslator.cpp]: sure, will re-add the mapping for backward compat

**Comment by phil65** [comment]
Cool. I'm still on holidays, so give me few more days.

**Comment by xhaggi** [comment]
addressed your comments.

File: addons/resource.language.en_gb/resources/strings.po
Comment (ksooo) [inline addons/resource.language.en_gb/resources/strings.po]: “/1080i/“ => “/xml/“

**Review by ksooo — APPROVED** [review]
just found another small glitch...

File: addons/resource.language.en_gb/resources/strings.po
Comment (xhaggi) [inline addons/resource.language.en_gb/resources/strings.po]: There are tons of other places references 1080i, we should change it at all.

Comment (ksooo) [inline addons/resource.language.en_gb/resources/strings.po]: Right, but what speaks against doing it right in the first place for new stuff we're adding? ;-)

Comment (xhaggi) [inline addons/resource.language.en_gb/resources/strings.po]: Nothing

Comment (ksooo) [inline addons/resource.language.en_gb/resources/strings.po]: :-)

**Comment by IchabodFletchman** [comment]
I've made the changes to Estuary including the new icon. @phil65 can cherry pick it once the PR is merged

**Comment by xhaggi** [comment]
@IchabodFletchman thanks for your contribution but we should think about changing the dialog to fit both, being an OSD dialog and a normal dialog laying on top of other dialogs. IMO the dialog looks a bit odd on top of the guide info dialog.

**Comment by xhaggi** [comment]
@phil65  do have some spare time to do the necessary skin changes?

**Comment by xhaggi** [comment]
I've dropped the skin changes for now because we can add it later.

@phil65 if you want to adapt something 8cf27edc39ad05da07be85497344de4b0ff2f41f or IchabodFletchman/skin.estuary@58333df

**Comment by xhaggi** [comment]
@ksooo I'll merge this after the build is fine.

**Comment by xhaggi** [comment]
jenkins build and merge

**Comment by xhaggi** [comment]
jenkins build and merge

**Comment by Leatherface75** [comment]
Would be nice with IMDB support too there.

**Pull request merged**
